packages/components/src/components/entities/actions.ts

```diff
@@ -317,18 +327,23 @@ export function getInitialParentChoices(parentTypeOptions: List<IEntityTypeOptio
     return parentValuesByType.sortBy(choice => choice.type.label, naturalSort).toList();
 }
 
-export function getUpdatedRowForParentChanges(parentDataType: EntityDataType, currentParents: List<EntityChoice>, childModel: QueryGridModel) {
+export function getUpdatedRowForParentChanges(parentDataType: EntityDataType, originalParents: List<EntityChoice>, currentParents: List<EntityChoice>, childModel: QueryGridModel) {
     const queryData = childModel.getRow();
     const queryInfo = childModel.queryInfo;
 
     let updatedValues = {};
-    currentParents.forEach((parentChoice) => {
-        // Label may seem wrong here, but it is the same as query when extracted from the original query to get
-        // the entity types.
-        if (parentChoice.value && parentChoice.value.length > 0) {
-            updatedValues[parentDataType.insertColumnNamePrefix + parentChoice.type.label] = parentChoice.value;
-        }
-    });
+    if (currentParents.isEmpty()) { // have no current parents but have original parents, send in empty strings so original parents are removed.
+        originalParents.forEach((parentChoice) => {
+            updatedValues[parentDataType.insertColumnNamePrefix + parentChoice.type.label] = "";
```

Comment:
I assume there is a reason, but why empty string instead of undefined or null?

Reply:
The empty string was consistent with treatment for other fields that are meant to be removed and I had tried using `undefined`, but it was completely stripped out.  Using null works, though, so that's changed now.